src/create-text-tracks-if-necessary.js

```diff
@@ -16,14 +16,34 @@ const createTextTracksIfNecessary = function(sourceBuffer, mediaSource, segment)
 
   // create an in-band caption track if one is present in the segment
   if (segment.captions &&
-      segment.captions.length &&
-      !sourceBuffer.inbandTextTrack_) {
-    removeExistingTrack(player, 'captions', 'cc1');
-    sourceBuffer.inbandTextTrack_ = player.addRemoteTextTrack({
-      kind: 'captions',
-      label: 'cc1'
-    }, false).track;
-    player.tech_.trigger({type: 'usage', name: 'hls-608'});
+      segment.captions.length) {
+    if (!sourceBuffer.inbandTextTracks_) {
+      sourceBuffer.inbandTextTracks_ = {};
+    }
+
+    for (let trackId in segment.captionTracks) {
+      let track = player.textTracks().getTrackById(trackId);
```

Comment:
Let's move this into the `if` block below. That way it will only go looking for the track on the player if the source buffer doesn't already have a `inbandTextTrack` stored for the `trackId`